internal/pkg/exec/docker.go

```diff
@@ -166,6 +168,33 @@ func (c DockerCommand) CheckDockerEngineRunning() error {
 	}
 }
 
+// GetPlatform will run the `docker version` command to get the OS/Arch.
+func (c DockerCommand) GetPlatform() (os, arch string, err error) {
+	if _, err := exec.LookPath("docker"); err != nil {
+		return "", "", ErrDockerCommandNotFound
+	}
+	buf := &bytes.Buffer{}
+	err = c.runner.Run("docker", []string{"version", "-f", "'{{json .Server}}'"}, Stdout(buf))
+	if err != nil {
+		return "", "", fmt.Errorf("run docker version: %w", err)
+	}
+	if c.buf != nil {
+		buf = c.buf
+	}
```

Comment:
I don't think we need this for testing, we should be able to mock `runner.Run` and check if the output `os, arch` is as expected